### DISCUSSION TITLE
Any benchmark of using ebpf map hash / array type from golang? is it fast?

### DISCUSSION THREAD

**Discussion by hiqsociety** [cilium/ebpf#624] [Q&A] [answered]
Any benchmark of using ebpf map hash / array type from golang? is it fast?

the call is it cgo or pure go? just wondering how much garbage is generated if using go and how fast it is if using cgo.

**Comment by lmb** [comment, marked as answer]
Hi, there should be some basic benchmarks mixed in with the tests. Feel free to take a look.

The library doesn't use cgo for syscalls, so it's probably going to be as fast as it will get. For simple map types like arrays syscall overhead will dominate the total time.

**Reply by hiqsociety** [reply]
>  syscall overhead will dominate the total time.
how expensive is syscall compared with accessing array in golang?
are there a lot of (or any) garbage generated with each syscall?

**Reply by lmb** [reply]
I don't have these numbers off the top of my head, I think you need to figure that out by yourself.
